### ISSUE TITLE
[Arm64] System.Security.Cryptography.Primitives.Tests GC hole

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/coreclr#17169]
```
xUnit.net console test runner (64-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Security.Cryptography.Primitives.Tests
Discovered:  System.Security.Cryptography.Primitives.Tests
Starting:    System.Security.Cryptography.Primitives.Tests

Assert failure(PID 16274 [0x00003f92], Thread: 17195 [0x432b]): !CREATE_CHECK_STRING(pMT && pMT->Validate())
    File: /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/object.cpp Line: 733
    Image: /home/sdmaclea/corefx/arm64-runtime/corerun


Assert failure(PID 16274 [0x00003f92], Thread: 17195 [0x432b]): !CREATE_CHECK_STRING(pMT && pMT->Validate())
    File: /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/object.cpp Line: 733
    Image: /home/sdmaclea/corefx/arm64-runtime/corerun

```

**Comment by sdmaclea** [comment]
Also `Security.Cryptography.OpenSsl.Tests` & `System.Security.Cryptography.Algorithms.Tests`

**Comment by sdmaclea** [comment]
#17168 & #17167 are likely GC holes too

**Comment by sdmaclea** [comment]
Fixed by #17536
